editor/project_manager.cpp

```diff
@@ -1627,40 +1627,28 @@ void ProjectManager::_show_project(const String &p_path) {
 	OS::get_singleton()->shell_open(String("file://") + p_path);
 }
 
-void ProjectManager::_scan_dir(DirAccess *da, float pos, float total, List<String> *r_projects) {
-
-	List<String> subdirs;
+void ProjectManager::_scan_dir(const String &path, List<String> *r_projects) {
+	DirAccess *da = DirAccess::create(DirAccess::ACCESS_FILESYSTEM);
+	da->change_dir(path);
 	da->list_dir_begin();
 	String n = da->get_next();
 	while (n != String()) {
 		if (da->current_is_dir() && !n.begins_with(".")) {
-			subdirs.push_front(n);
+			_scan_dir(da->get_current_dir() + "/" + n, r_projects);
```

Comment:
Wouldn't appending use less memory and take less execution time than concatenating?

Reply:
I'm not sure if it really matters much. One things though, you could use `String::plus_file()` to append the filename to the dir path instead of concatenating `"/"` manually (which could result in `//` in the path).

Reply:
Is this really useful since I don't think it's actually possible to end up with `//`. Additionally `//` and `/` behave the same way normally (I don't know if this is important in Godot).

Wouldn't this also mean that we should use it here:
https://github.com/godotengine/godot/blob/252c841d7ff5d770e8d2819e4f7955363410744b/core/ustring.cpp#L3891

Reply:
It's mostly a style/consistency thing, most of the time a `//` is not a problem, so it can be awkward on paths that will end up converted with `replace("/", "\\")` on Windows.

> Wouldn't this also mean that we should use it here:

Well that's where `plus_file` is implemented... So no, unless we want an infinite recursion :)